oci-devops-cicd-helm-ci/helm-deployment/helm-deployment.md

```diff
@@ -0,0 +1,226 @@
+
+![oci-helm-build-pipelines.png](images/oci-helm-build-pipelines.png)
+
+1. click `Parameters` and edit the value for `GPG_ARTIFACT_OCID` using the `pen` icon.
+
+![oci-build-params.png](images/oci-build-params.png)
+
+1. Update the value with `OCID of Artifacts uploaded` and use the `tick` symbol and save it.
+
+![](images/oci-build-params-edit.png)
+
+1. Switch back to `build a pipeline` and click on `Start manual run` and start a building job.
+
+![oci-build-manual-run.png](images/oci-build-manual-run.png)
+
+1. This will start the build job. While waiting for the job to complete, switch back to the `Build pipeline` tab, Click `View details` under managed build stage.
+
+![oci-managed-build-stage-details.png](images/oci-managed-build-stage-details.png)
+
+1. It will show the custom resource setups for the builder machines.
+
+![oci-build-custom-shapes.png](images/oci-build-custom-shapes.png)
+
+1. Close the tab, Switch to `Build history and follow the current ongoing tasks and wait for it to finish.
+
+![oci-build-done.png](images/oci-build-done.png)
+
+1. Switch to `DevOps Project Resources` > `Deployment pipelines`
+
+![oci-devops-pipelines.png](images/oci-devops-pipelines.png)
+
+1. Click the pipeline named `deploy_helmchart_<ID>`.In the pipeline use `3 dots` and view details of the helm chart stage.
+
+![oci-helm-stage-details.png](images/oci-helm-stage-details.png)
+
+1. You can verify the different helm options and dynamic values set for the deployments.
+
+![oci-helm-options.png](images/oci-helm-options.png)
+
+1. Close and click on the `Deployments` tab and open the recent deployments done or ongoing.
+
+![oci-deployment-recent.png](images/oci-deployment-recent.png)
+
+1. Click on the deployment and validate the stages and logs.
+
+![](images/oci-helm-deployment-logs.png)
+
+1. Search `OKE` in the search bar and open the OKE Cluster to view the deployment.
+
+![oci-oke-cluster-view.png](images/oci-oke-cluster-view.png)
+
+1. Click on the `Access Cluster` button.
+
+![oci-oke-access-cluster.png](images/oci-oke-access-cluster.png)
+
+1. Preferably use `Cloud Shell Access` and set up access to your OKE Cluster.
+
+![oci-oke-cs-access.png](images/oci-oke-cs-access.png)
+
+1. On the cloud shell run `kubectl get service/ocidevops-node-service` and fetch the Load Balancer IP address.
+
+![oci-kubec-lb-ip.png](images/oci-kubec-lb-ip.png)
+
+1. Launch the application by using the URL `http://LoadBalancer IP`
+
+![oci-app-url.png](images/oci-app-url.png)
+
+1. Optionally you can update the helm chart of the application and re-deploy the application, by running a manual build once the updated code is pushed back to the code repo.
+
+You may now **proceed to the next lab**.
+
+## Learn More
+
+* [OCI Devops documentation](https://docs.oracle.com/en-us/iaas/Content/devops/using/home.htm)
+
+
+## Acknowledgements
+
+* **Author** - Rahul M R
+* **Contributors** -
```

Comment:
@RahulMR42 - Add contributor details. If there are no contributors, you can remove that line.

Reply:
Done